pkg/providers/cloudstack/config/template-cp.yaml

```diff
@@ -58,50 +74,59 @@ kind: CloudStackMachineTemplate
 metadata:
   name: {{.controlPlaneTemplateName}}
   namespace: {{.eksaSystemNamespace}}
-{{- if .cloudstackControlPlaneAnnotations}}
+{{- if $annotationsCP}}
   annotations:
 {{- end }}
-{{- if .cloudstackControlPlaneDiskOfferingProvided}}
-    mountpath.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingPath}}
-    device.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingDevice}}
-    filesystem.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingFilesystem}}
-    label.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingLabel}}
+{{- if $diskConfiguredCP}}
+    diskoffering.{{$suffix}}: "{{.cloudstackControlPlaneDiskOfferingJsonStr | js}}"
```

Comment:
what does `| js` do? Can we use this tool to avoid passing in the redundant values to the templater?

Reply:
it transfer {"a" : "b"} into {\"a\": \"b\"}

Since annotation is map[string]string, and " is used as start/end of a string, so we have to | js to escape "